code/game/objects/items/devices/advnifrepair.dm

```diff
@@ -24,13 +24,15 @@
 /obj/item/device/nifrepairer/attackby(obj/W, mob/user)
 	if(istype(W,/obj/item/stack/nanopaste))
 		var/obj/item/stack/nanopaste/np = W
-		if(np.use(1) && supply.get_free_space() >= efficiency)
```

Comment:
A better way to achieve that would be to swap the order of this. Put the check for us being empty first, and use() second. That way it will only attempt to use if first check has passed successfully.

Reply:
Done as requested.